src/properties/components/Properties.tsx

```diff
@@ -50,28 +70,28 @@ export const Properties = () => {
     () => tryFindByUUID(uiSchema, selection?.uuid),
     [selection, uiSchema]
   );
-  const canSetUISchemaOptions = (
-    uiElement: EditorUISchemaElement | undefined,
-    updatedProperties: any
-  ): boolean =>
-    !!uiElement &&
-    !isEqual(updatedProperties, uiElement.options) &&
-    !(uiElement.options === undefined && isEmpty(updatedProperties));
 
   const updateProperties = useCallback(
     ({ data: updatedProperties }) => {
-      if (canSetUISchemaOptions(uiElement, updatedProperties)) {
+      if (canUpdateUiSchemaElement(uiElement, updatedProperties)) {
         dispatch(Actions.setUiSchemaOptions(uiElement, updatedProperties));
       }
     },
     [dispatch, uiElement]
   );
+
+  const getData = useCallback(
+    (propertiesSchema: JsonSchema) =>
+      propertiesService.getDataForProperties(uiElement, propertiesSchema),
+    [uiElement]
+  );
+
   if (!selection) return <NoSelection />;
 
   const properties = getProperties(uiElement, schema);
   return properties ? (
     <JsonForms
-      data={uiElement.options}
+      data={getData(properties.schema)}
```

Comment:
why scope down the data?

Reply:
Not necessary, replaced with clone of uiElement